apps/agreement/contracts/Agreement.sol

```diff
@@ -698,6 +707,28 @@ contract Agreement is IAgreement, AragonApp {
 
     // Internal fns
 
+    function _payTransactionFees(uint256 settingId, IDisputable _disputable, Staking _staking, address _submitter) internal {
+        // Get fees
+        IArbitrator arbitrator = settings[settingId].arbitrator;
+        address transactionFeesOracleAddress = arbitrator.getModule(TRANSACTION_FEES_MODULE);
+        require(transactionFeesOracleAddress != address(0), ERROR_TX_FEES_MODULE_NOT_FOUND);
+        ITransactionFeesOracle transactionFeesOracle = ITransactionFeesOracle(transactionFeesOracleAddress);
+        bytes32 appId = _disputable.appId();
+        (ERC20 token, uint256 amount, address beneficiary) = transactionFeesOracle.getFee(appId);
+
+        if (amount == 0) {
+            return;
+        }
+
+        // Pay fees
+        if (token == _staking.token()) {
```

Comment:
I don’t get this one. `_staking` is coming from the cached staking pool in `requirement`. How is this different to what we do in `_lockBalance` call inside `newAction`?

Reply:
Since we need to call `stakingFactory.getOrCreateInstance(token)` (a second call) in the worst case, would it be an optimization to just always call it (so it's always one call)?